app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -16,6 +16,67 @@ import scala.concurrent.{ExecutionContext, Future, Promise}
 /** Useful utilities to use in the wallet project for syncing things against bitcoind */
 object BitcoindRpcBackendUtil extends BitcoinSLogger {
 
+  /** Has the wallet process all the blocks it has not seen up until bitcoind's chain tip */
+  def catchupWalletToBitcoind(bitcoind: BitcoindRpcClient, wallet: Wallet)(
+      implicit ec: ExecutionContext): Future[Unit] = {
+
+    def doSync(walletHeight: Int, bitcoindHeight: Int): Future[Unit] = {
+      if (walletHeight > bitcoindHeight) {
+        Future.failed(new RuntimeException(
+          s"Bitcoind and wallet are in incompatible states, " +
+            s"wallet height: $walletHeight, bitcoind height: $bitcoindHeight"))
+      } else {
+        val blockRange = walletHeight.to(bitcoindHeight).tail
+
+        logger.info(s"Syncing ${blockRange.size} blocks")
+
+        val func: Vector[Int] => Future[Unit] = { range =>
+          val hashFs =
+            range.map(bitcoind.getBlockHash(_).map(_.flip))
+          for {
+            hashes <- Future.sequence(hashFs)
+            _ <- wallet.nodeApi.downloadBlocks(hashes)
+          } yield ()
+        }
+
+        FutureUtil
+          .batchExecute(elements = blockRange.toVector,
+                        f = func,
+                        init = Vector.empty,
+                        batchSize = 25)
+          .map(_ => ())
+      }
+    }
+
+    for {
+      bitcoindHeight <- bitcoind.getBlockCount
+      walletStateOpt <- wallet.getSyncHeight()
+      _ <- walletStateOpt match {
+        case None =>
+          for {
+            utxos <- wallet.listUtxos()
+            lastConfirmedOpt = utxos.filter(_.blockHash.isDefined).lastOption
+            _ <- lastConfirmedOpt match {
+              case None => FutureUtil.unit
```

Comment:
So if we shutdown the bitcoin-s wallet in a state that only has unconfirmed utxos, we won't be able to sync ever?

Reply:
Yeah I guess, not sure the best way to solve this, only real solution would be to rescan every block.